### PR TITLE
Presentation

### PR DESCRIPTION
**Pull request by Robert-Kolmos**
- Implement StepViews for ResearchStack
- Add ResearchStack ShowStepViewModels and Factories
- Refactor StepMapper.

### CODE REVIEW

File: presentation/src/main/java/org/sagebionetworks/research/presentation/DisplayString.java [hunk 1]
```diff
@@ -32,20 +32,72 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
 import android.support.annotation.Nullable;
 import android.support.annotation.StringRes;
 
-public class DisplayString {
+import com.google.common.base.Objects;
+
+
+public class DisplayString implements Parcelable {
```
Comment (liujoshua) [inline presentation/src/main/java/org/sagebionetworks/research/presentation/DisplayString.java]: Is there a reason we're not using @AutoValue for the parcelable? We should write unit tests if we're implementing it ourselves

File: presentation/src/main/java/org/sagebionetworks/research/presentation/model/form/ChoiceInputFieldBase.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.google.common.collect.Range;
+
+import org.sagebionetworks.research.domain.form.DataTypes.InputDataType;
+import org.sagebionetworks.research.domain.form.TextField.TextFieldOptions;
+import org.sagebionetworks.research.domain.form.interfaces.SurveyRule;
+import org.sagebionetworks.research.presentation.DisplayString;
+
+import java.util.List;
+
+public class ChoiceInputFieldBase<E extends Parcelable> extends InputFieldViewBase {
```
Comment (liujoshua) [inline presentation/src/main/java/org/sagebionetworks/research/presentation/model/form/ChoiceInputFieldBase.java]: Will E extends Parcelable handle our use cases? I'm thinking things like Choice<Double> Choice<String>

We can leave it for now and remove `extends Parcelable` if that turns out better

File: presentation/src/main/java/org/sagebionetworks/research/presentation/model/implementations/ActiveUIStepViewBase.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        return isBackgroundAudioRequired;
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
```
Comment (liujoshua) [inline presentation/src/main/java/org/sagebionetworks/research/presentation/model/implementations/ActiveUIStepViewBase.java]: Do we have a test which covers this?

File: presentation/src/main/java/org/sagebionetworks/research/presentation/DisplayString.java (hunk 1 above)
Comment (Robert-Kolmos) [reply presentation/src/main/java/org/sagebionetworks/research/presentation/DisplayString.java]: I didn't implement it I just used AndroidStudio code generation to do it, however I agree I will switch it to an autovalue class

File: presentation/src/main/java/org/sagebionetworks/research/presentation/model/form/ChoiceInputFieldBase.java (hunk 2 above)
Comment (Robert-Kolmos) [reply presentation/src/main/java/org/sagebionetworks/research/presentation/model/form/ChoiceInputFieldBase.java]: I think if we don't make how to write the overall input field to a parcel. I might be missing something here.

File: presentation/src/main/java/org/sagebionetworks/research/presentation/model/implementations/ActiveUIStepViewBase.java (hunk 3 above)
Comment (Robert-Kolmos) [reply presentation/src/main/java/org/sagebionetworks/research/presentation/model/implementations/ActiveUIStepViewBase.java]: I don't believe we do I will go ahead and write some more tests for the step views.

**Pull request merged**